server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1487,6 +1498,15 @@ private void cleanupStaleIndices(
         }
     }
 
+    private static boolean isIndexWithIndexUUIDPresent(ClusterService clusterService, String indexUUID) {
+        for (final IndexMetadata indexMetadata : clusterService.state().metadata().getIndices().values()) {
```

Comment:
instead of iterating over all indices , we can directly get the metadata for that index name  

```
IndexMetadata indexMetadata = clusterState.metadata().getIndices().get(indexId);
 if (indexUUID.equals(indexMetadata.getIndexUUID()))
{
//cleanup
}
```
```

Reply:
yeah.. we don't have indexName here, to get indexName, we need to read index metadata file, instead, i am just interating over all indices to get the index we need.